plugins/woocommerce-admin/client/products/utils/get-product-stock-status.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { PartialProduct, ProductVariation } from '@woocommerce/data';
+
+/**
+ * Labels for product stock statuses.
+ */
+export enum PRODUCT_STOCK_STATUS_KEYS {
+	instock = 'instock',
+	onbackorder = 'onbackorder',
+	outofstock = 'outofstock',
+}
+
+/**
+ * Labels for product stock statuses.
+ */
+export const PRODUCT_STOCK_STATUS_LABELS = {
+	[ PRODUCT_STOCK_STATUS_KEYS.instock ]: __( 'In stock', 'woocommerce' ),
+	[ PRODUCT_STOCK_STATUS_KEYS.onbackorder ]: __(
+		'On backorder',
+		'woocommerce'
+	),
+	[ PRODUCT_STOCK_STATUS_KEYS.outofstock ]: __(
+		'Out of stock',
+		'woocommerce'
+	),
+};
+
+/**
+ * Get the product stock quantity or stock status label.
+ *
+ * @param  product Product instance.
+ * @return {PRODUCT_STOCK_STATUS_KEYS} Product staus key.
```

Comment:
Two nitpick things:
- There is a typo in `staus`.
- Since we can also return a number, I think that this line should be like this:
```
@return {PRODUCT_STOCK_STATUS_KEYS|number} Product stock quantity or product status key.
```